source/agora/common/crypto/Key.d

```diff
@@ -112,17 +152,32 @@ public struct PublicKey
         return PublicKey(typeof(this.data)(bin[1 .. $ - 2]));
     }
 
-    /// Expose verification capability
-    public const(ubyte)[] verify (scope const(ubyte)[] msg) const
+    ///
+    unittest
     {
-        if (msg.length < crypto_sign_ed25519_BYTES)
-            return null;
+        immutable address = `GDD5RFGBIUAFCOXQA246BOUPHCK7ZL2NSHDU7DVAPNPTJJKVPJMNLQFW`;
+        PublicKey pubkey = PublicKey.fromString(address);
+        assert(pubkey.toString() == address);
+    }
+
+    /***************************************************************************
+
+        Verify that a signature matches a given message
 
-        ubyte[] res = new ubyte[](msg.length - crypto_sign_ed25519_BYTES);
-        size_t reslen;
-        if (crypto_sign_ed25519_open(res.ptr, &reslen, msg.ptr, msg.length, this.data[].ptr) != 0)
-            return null;
-        return res[0 .. reslen];
+        Params:
+          signature = The signature of `msg` matching `this` public key.
```

Comment:
I've just noticed you use two spaces here. I haven't even thought about this, I just tab it always (so 4). Is this part of the style guide?

Reply:
Hum habits I guess. 2 or 4 works for me, I just need to get used to it.